agent/jvm/src/main/java/org/jolokia/jvmagent/security/asn1/DERInteger.java

```diff
@@ -0,0 +1,99 @@
+/*
+ * Copyright 2009-2021 Roland Huss
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *       http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.jolokia.jvmagent.security.asn1;
+
+public class DERInteger implements DERObject {
```

Comment:
Are those ASN1 objects copied over from some other open-source library (like bouncycastle) or do you have written those ? 

It's probably over the top here, but are there any tests for those classes ?

Reply:
The tests are in `org/jolokia/jvmagent/security/asn1/DEREncodingTest.java` and I confirm - I looked a bit at BC classes (only to check what _DER Tag_ is), but every line of code is written by me. I was consulting only the X.680/X.690 specifications.